### DISCUSSION TITLE
New indicator not importing.

### DISCUSSION THREAD

**Discussion by Alpha23-Ltd** [nautechsystems/nautilus_trader#1055] [General]
Hi, 

I have created a new indicator (see code below), both the .pyx and .pyd files.
I do `make build` after this and the .so and .c files are created. So Cythonize seems to be working. There are no error messages.

In my strategy .py file I am not able to import the Indicator as per this line:

`from nautilus_trader.indicators.weighted_mid import WeightedMid`

I dont think it's a problem with the code, since even if i duplicate an existing indicator and rename it, it still wont import.
Do the Cython  indicators need to be declared elsewhere?

thanks

`from nautilus_trader.core.correctness cimport Condition
from nautilus_trader.indicators.base.indicator cimport Indicator
from nautilus_trader.model.data.tick cimport QuoteTick
from nautilus_trader.model.identifiers cimport InstrumentId
from nautilus_trader.model.objects cimport Price

cdef class WeightedMid(Indicator):
    """
    Provides a weighted mid price by quantities on top level.

    Parameters
    ----------
    instrument_id : InstrumentId
        The instrument ID for the tick updates.

    Raises
    ------
    ValueError
        If `capacity` is not positive (> 0).
    """

    def __init__(self, InstrumentId instrument_id not None) -> None:
        super().__init__(params=[instrument_id])

        self.instrument_id = instrument_id
        self.mid = 0.0

    cpdef void handle_quote_tick(self, QuoteTick tick) except *:
        """
        Update with the given quote tick.

        Parameters
        ----------
        tick : QuoteTick
            The tick for the update.

        Raises
        ------
        ValueError
        """
        Condition.not_none(tick, "tick")
        Condition.equal(self.instrument_id, tick.instrument_id, "instrument_id", "tick.instrument_id")

        # Check initialization
        if not self.initialized:
            self._set_has_inputs(True)

        cdef double ask = Price.raw_to_f64_c(tick._mem.ask.raw)
        cdef double bid = Price.raw_to_f64_c(tick._mem.bid.raw)
        cdef double ask_size = Price.raw_int64_c(tick._mem.ask_size.raw)
        cdef double bid_size = Price.raw_int64_c(tick._mem.bid_size.raw)

        cdef double weighted_mid = (bid * ask_size + ask * bid_size) / (bid_size + ask_size)

        self.mid = weighted_mid

    cpdef void _reset(self) except *:
        self.mid = 0
`

**Comment by cjdsellers** [comment]
I see the issue is closed, were you able to solve the problem in the end?

**Reply by Alpha23-Ltd** [reply]
Yes. The autocomplete in Pycharm made it look like it wasn't importing but it did once I referenced the module in the code.

Thanks

**Comment by cjdsellers** [comment]
Cython doesn't seem to play so nicely with PyCharm at times, particularly imports and definitions.
